tests/tests/bitasset_tests.cpp

```diff
@@ -1403,4 +1403,241 @@ BOOST_AUTO_TEST_CASE(hf_890_test_hf1270)
 
 } FC_LOG_AND_RETHROW() }
 
+int num_limit_orders_on_books( graphene::chain::database& db )
+{
+   const auto& limit_index = db.get_index_type<limit_order_index>().indices().get<by_price>();
+   int count = 0;
+   std::for_each( limit_index.begin(), limit_index.end(), [&count](const limit_order_object& obj) {
+      count++; 
+   });  
+   return count; 
+}
+
+int num_call_orders_on_books( graphene::chain::database& db )
+{
+   const auto& call_index = db.get_index_type<call_order_index>().indices().get<by_collateral>();
+   int count = 0;
+   std::for_each( call_index.begin(), call_index.end(), [&count](const call_order_object& obj) {
+      std::stringstream output;
+      output << "Call order: " << obj.id.number
+            << " is for " << obj.get_collateral().amount.value << " of asset id " << obj.get_collateral().asset_id.instance.value
+            << " which is collateralizing a debt of " << obj.get_debt().amount.value << " of asset id " << obj.get_debt().asset_id.instance.value
+            << "\n   Execution price is " << obj.call_price.base.amount.value << " of asset " << obj.call_price.base.asset_id.instance.value
+            << " versus " << obj.call_price.quote.amount.value << " of asset " << obj.call_price.quote.asset_id.instance.value;
+      BOOST_TEST_MESSAGE( output.str() );
+      count++; 
+   }); 
+   return count;  
+}
+
```

Comment:
// create an asset issuer charlie, who will also trade
// create three price feed producers: feeder1, feeder2, feeder3
// create two traders alice and bob
// alice will also become a debt holder of the MPA JMJCOIN